db/00000000000003_outstanding_batch_state.up.sql

```diff
@@ -0,0 +1,8 @@
+-- Specifies the possible state of an outstanding batch.
+CREATE TYPE OUTSTANDING_BATCH_STATE AS ENUM(
+    'FILLING',  -- this outstanding batch is still being considered for additional reports
+    'FILLED'    -- this outstanding batch has received enough reports, no more are necessary
+);
+ALTER TABLE outstanding_batches ADD COLUMN state OUTSTANDING_BATCH_STATE NOT NULL DEFAULT 'FILLING';
+CREATE INDEX outstanding_batches_task_id_and_time_bucket_start ON outstanding_batches(task_id, time_bucket_start) WHERE state = 'FILLING';
```

Comment:
Hmm. My first thought is that we may want to CREATE INDEX CONCURRENTLY to persuade postgres to take more lenient locks, but then again aggregation job processing is non-critical.